test/controllers/enquiry.controller.test.js

```diff
@@ -606,22 +656,132 @@ describe('Enquiry Controller', () => {
+      await addUser(demoUser);
+      await addProperty(vendor2Property);
+      await addEnquiry(vendorEnquiry);
+      await addEnquiry(vendor2Enquiry);
     });
 
-    context('with a valid token & id', () => {
-      it('returns successful payload', (done) => {
-        request()
-          .get(`/api/v1/enquiry/${enquiry._id}`)
-          .set('authorization', adminToken)
-          .end((err, res) => {
-            expect(res).to.have.status(200);
-            expect(res.body.success).to.be.eql(true);
-            expect(res.body).to.have.property('enquiry');
-            expect(res.body.enquiry._id).to.be.eql(enquiry._id.toString());
-            done();
-          });
+    context('with admin token ', () => {
+      [vendorEnquiry, vendor2Enquiry].map((enquiry) =>
+        it('returns successful payload', (done) => {
+          request()
+            .get(`/api/v1/enquiry/${enquiry._id}`)
+            .set('authorization', adminToken)
+            .end((err, res) => {
+              expect(res).to.have.status(200);
+              expect(res.body.success).to.be.eql(true);
+              expect(res.body).to.have.property('enquiry');
+              expect(res.body.enquiry._id).to.be.eql(enquiry._id.toString());
+              done();
+            });
+        }),
+      );
+    });
+
+    context('with a token with invalid access', () => {
+      // eslint-disable-next-line array-callback-return
+      [vendorUser, regularUser].map((user) => {
+        let token;
+        beforeEach(async () => {
+          const loggedInUser = await loginUser(user);
+          token = loggedInUser.token;
+        });
+
+        it('returns not found', (done) => {
+          request()
+            .get(`/api/v1/enquiry/${vendor2Enquiry._id}`)
+            .set('authorization', token)
+            .end((err, res) => {
+              expect(res).to.have.status(404);
+              expect(res.body.success).to.be.eql(false);
+              expect(res.body.message).to.be.eql('Enquiry not found');
+              done();
+            });
+        });
+      });
+    });
+
+    context('with a token with valid access', () => {
+      // eslint-disable-next-line array-callback-return
+      [demoUser, vendor2].map((user) => {
+        let token;
+        beforeEach(async () => {
+          const loggedInUser = await loginUser(user);
+          token = loggedInUser.token;
+        });
+
+        it('returns enquiry', (done) => {
+          request()
+            .get(`/api/v1/enquiry/${vendor2Enquiry._id}`)
+            .set('authorization', token)
+            .end((err, res) => {
+              expect(res).to.have.status(200);
+              expect(res.body.success).to.be.eql(true);
+              expect(res.body).to.have.property('enquiry');
+              expect(res.body.enquiry._id).to.be.eql(vendor2Enquiry._id.toString());
+              done();
+            });
+        });
+      });
+    });
+
+    context('with a token with invalid access', () => {
```

Comment:
The token actually has a valid access but they are not authorized to view another user's enquiry.

```suggestion
    context('when viewing an unauthorized enquiry', () => {
```